File: C++ for Everyone, 2nd Ed, Horstmann/Chapter3/P3-7.cpp
//Write a program that reads three numbers and prints “ increasing ” if they are in
//increasing order, “ decreasing ” if they are in decreasing order, and “ neither ” other -
//wise.Here, “increasing” means “strictly increasing”, with each value larger than its
//pre decessor.The sequence 3 4 4 would not be considered increasing.

#include<iostream>

using namespace std;

int main()
{
    cout << "Enter three numbers: ";
    float n1, n2, n3;
    cin >> n1 >> n2 >> n3;

    if (n1 < n2 && n2 < n3 )
    {
        cout << "Increasing order.\n";
    }

    else if (n1 > n2 && n2 > n3)
    {
        cout << "Decreasing order.\n";
    }

    else
    {
        cout << "Neither.\n";
    }

    return 0;
}

File: Python For Everyone, Horstmann/Chapter3/P3.7.py
# Write a program that reads in three integers and prints “in order” if they are sorted in
# ascending or descending order, or “not in order” otherwise. For example,
#     1 2 5 in order
#     1 5 2 not in order
#     5 2 1 in order
#     1 2 2 in order

number1 = float(input("Enter number one: "))
number2 = float(input("Enter number two: "))
number3 = float(input("Enter number three: "))

if (number1 < number2 < number3) or (number1 > number2 > number3)\
    or (number1 <= number2 < number3) or (number1 < number2 <= number3)\
    or (number1 >= number2 > number3) or (number1 > number2 >= number3):
    print("In order")

else:
    print("Not in order")